src/bdc/steps/gpt_summarizer.py

```diff
@@ -0,0 +1,125 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Berkay Bozkurt <resitberkaybozkurt@gmail.com>
+# SPDX-FileCopyrightText: 2023 Sophie Heasman <sophieheasmann@gmail.com>
+
+
+from http import HTTPStatus
+
+import openai
+import pandas as pd
+import requests
+from bs4 import BeautifulSoup
+from pandas import DataFrame
+from requests import RequestException
+from tqdm import tqdm
+
+from bdc.steps.step import Step, StepError
+from config import OPEN_AI_API_KEY
+
+
+class GPTSummarizer(Step):
+    name = "GPT-Summarizer"
+    model = "gpt-4"
+    no_answer = "None"
+
+    # system and user messages to be used for creating company summary for lead using website.
+    system_message_for_website_summary = f"You are html summarizer, you being provided the companies' htmls and you answer with the summary of three to five sentences including all the necessary information which might be useful for salesperson. If no html then just answer with '{no_answer}'"
+    user_message_for_website_summary = (
+        "Give salesperson a summary using following html: {}"
+    )
+
+    extracted_col_name_website_summary = "sales_person_summary"
+    client = None
+
+    def load_data(self) -> None:
+        self.client = openai.OpenAI(api_key=OPEN_AI_API_KEY)
+
+    def verify(self) -> bool:
+        if OPEN_AI_API_KEY is None:
+            raise StepError("An API key for openAI is need to run this step!")
+        return self.df is not None and "google_places_website" in self.df
+
+    def run(self) -> DataFrame:
+        tqdm.pandas(desc="Summarizing the website of leads")
+        self.df[self.extracted_col_name_website_summary] = self.df[
+            "google_places_website"
+        ].progress_apply(lambda lead: self.summarize_the_company_website(lead))
```

Comment:
For some reason it doesn't seem to take axis as a parameter, but seems to work without. We discussed this the other day, and think that it automatically sets axis as 1. @rbbozkurt do you have any thoughts?

Reply:
I think it works because you select only the column "google_places_website" which only has one axis, which is row by row.  
Alternative would be like this:
```suggestion
       self.df[self.extracted_col_name_website_summary] = self.df.progress_apply(lambda lead:
           self.summarize_the_company_website(lead["google_places_website"]), axis=1)
```

Reply:
Yep I see, changed to your suggestion for consistency with the other steps